### PR TITLE
Fixes #11658 - when adding/removing puppetclasses to host/hostgroup, …

### PR DESCRIPTION
**Pull request by xprazak2**
…list of puppet class params gets re-rendered as a result of call to the server

### CODE REVIEW

File: app/helpers/hosts_and_hostgroups_helper.rb [hunk 1]
```diff
@@ -102,12 +102,12 @@ def realm_field(f, can_override = false, override = false)
 
   def interesting_klasses(obj)
     classes    = obj.all_puppetclasses
-    classes_ids = classes.reorder('').pluck('puppetclasses.id')
+    classes_ids = (classes.reorder('').pluck('puppetclasses.id') + obj.puppetclasses.map(&:id)).uniq.sort
```
Comment (ohadlevy) [inline app/helpers/hosts_and_hostgroups_helper.rb:105]: why do we need to sort ids?

Comment (xprazak2) [inline app/helpers/hosts_and_hostgroups_helper.rb:105]: @ohadlevy: for the same reason we call `reorder`? I am convinced we do not need it but there must be some reason it is there, perhaps I just do not see it...

**Comment by domcleal** [comment]
Tests are red, please check them.

File: app/helpers/hosts_and_hostgroups_helper.rb (hunk 1 above)
Comment (ohadlevy) [inline app/helpers/hosts_and_hostgroups_helper.rb:105]: On Sep 9, 2015 10:51, "xprazak2" notifications@github.com wrote:

> In app/helpers/hosts_and_hostgroups_helper.rb:
> 
> > @@ -102,12 +102,12 @@ def realm_field(f, can_override = false, override
> > = false)
> > 
> >    def interesting_klasses(obj)
> >      classes    = obj.all_puppetclasses
> > -    classes_ids = classes.reorder('').pluck('puppetclasses.id')
> > -    classes_ids = (classes.reorder('').pluck('puppetclasses.id') +
> >   obj.puppetclasses.map(&:id)).uniq.sort
> 
> @ohadlevy: for the same reason we call reorder? I am convinced we do not
> need it but there must be some reason it is there, perhaps I just do not
> see it...

I think we remove ordering to make the db query faster as we don't care
about ordering by name for example.
Further pg will complain if you order by a field which is not in the select
statement.

Comment (xprazak2) [inline app/helpers/hosts_and_hostgroups_helper.rb:105]: @ohadlevy: thank you for explanation, I will remove `sort`

**Comment by xprazak2** [comment]
I will be deleting this branch, it was decided we will go another way ([details](https://github.com/theforeman/foreman/pull/2658#issuecomment-139242017))

**Pull request closed without merging**